js/controllers/myStatisticsCtrl.js

```diff
@@ -424,6 +448,91 @@ GdeTrackingApp.controller("myStatisticsCtrl",					function($scope,	$location,	$h
     toggleDialog('selectProductGroups');
   };
 
+  $scope.showDelTrashDialog= function(activityId){
+    $scope.delTrashActId = activityId;
+    toggleDialog('delTrashDialog');
+  }
+
+  $scope.trashGDEActivity = function(){
+    var activityId = $scope.delTrashActId;
+
+    //Remove the AR from the backend
+    $scope.gdeTrackingAPI.activity_record.trash({id:activityId}).execute(
+      function(resp){
+        if (resp.code){
+          console.log('gdeTrackingAPI.activity_record.trash({id:'+activityId+'}) responded with Response Code: '+resp.code + ' - '+ resp.message);
+
+          alert(resp.message);
+        }else{
+          toggleDialog('delTrashDialog');
+
+          //Get the item
+          var item = $.grep($scope.data.items, function(item){
+            return item.id== activityId;
+          })[0];
+
+          item.deleted=true;
+          //remove the item from the array
+          removeARfromList(resp.id);
+
+          //Re push updated Item only if the user is displaying the trashed items
+          if ($scope.includeDeleted==true){
```

Comment:
This check is not needed after last changes to avoid calling the API back every time the trash flag is toggled.
The updated item should be re added to the list every time